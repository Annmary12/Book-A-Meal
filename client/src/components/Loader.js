import React from 'react';
import PropTypes from 'prop-types';
import loadingCube from '../../assets/img/loading-cube.svg';

const Loader = ({ width, message }) => (
  <div style={{ textAlign: 'center' }}>
    <img src={loadingCube} width={width} alt="loading" /><br />
    { message }
  </div>
);

Loader.propTypes = {
  width: PropTypes.number,
  message: PropTypes.string
};

Loader.defaultProps = {
  width: 40,
  message: ''
};
export default Loader;
