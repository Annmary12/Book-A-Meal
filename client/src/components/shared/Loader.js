import React from 'react';
import PropTypes from 'prop-types';
import loadingCube from '../../../assets/img/loading-cube.svg';

const Loader = ({ width }) => (
  <div style={{ textAlign: 'center' }}>
    <img src={loadingCube} width={width} alt="loading" /><br />
    loading...
  </div>
);

Loader.propTypes = {
  width: PropTypes.number
};

Loader.defaultProps = {
  width: 40
};
export default Loader;
