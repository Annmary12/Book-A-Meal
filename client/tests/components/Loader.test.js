import React from 'react';
import Loader from '../../src/components/Loader.js';

describe("Loader component", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
  })
})