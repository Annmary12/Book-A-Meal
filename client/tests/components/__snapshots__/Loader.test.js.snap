// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should match snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <img
    alt="loading"
    src="test-file-stub"
    width={40}
  />
  <br />
  loading...
</div>
`;
